### PR TITLE
Fix signing in for users with non-default locale

### LINKED ISSUE #13082: Account settings language not used after login - requires browser refresh to fully translate UI
**Describe the bug**
After setting up the language in "Account settings", if you log out and log in back again Metabase remembers your language but shows everything in English (except for the word "table", weirdly enough 👀).

Workaround is to refresh the page and then everything is translated again.

**Logs**
When redirected to the login page, the moment app refreshes into English, this error shows up:
`app-main.bundle.js?d4167a17e88b8bf103b8:5 GET http://localhost:13981/api/user/current 401 (Unauthorized)`

**To Reproduce**
1. Go to "Account settings" (gear icon > Account settings)
2. Choose language other than English
3. Click "Update" button
4. Observe that the interface updated and everything is now translated
5. Click on "gear" icon ("Settings")
6. Choose the last/bottom option ("Sign out")
7. You should be redirected to the login page `/auth/login?redirect=%2F`
8. Observe that after a second or so, the page updates and the translations are gone (it is shown in English again)
9. Log in
10. Everything is shown in English, except the translation in x-rays for the word "table"
11. Go to "Account settings" again and you'll see that your language is still selected
12. Refresh the page and translations for your chosen language are back

**Expected behavior**
Metabase should persist/remember user locale settings across sessions.

**Screenshots**
https://share.getcloudapp.com/kpubdEy6 (video recording)

**Information about your Metabase Installation:**
This happens on local dev (v0.37.0-snapshot), Docker latest (v0.36.3) and Mac app (v0.36.0).

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: any browser
- Your operating system: OS X 10.15.6
- Your databases: N/A
- Metabase version: multiple (see above)
- Metabase hosting environment: multiple (see above)
- Metabase internal database: H2 (default)

**Severity**
Since there is a fairly simple workaround, it's not severe at all, but I can imagine how it can be annoying and/or confusing for users that are not using English as their chosen language.

**Additional context**
Originally documented in #13063

### LINKED ISSUE #16532: Logout button brings user back to main page momentarily before logging out
**Describe the bug**
When you click the logout button, you're briefly brought back to the main (post-login) screen before being brought to the logout screen. This delay happens on both Firefox and Chrome but is significantly longer on Firefox (see video). It's confusing since you're momentarily unsure whether the logout request took effect.

**Expected behavior**
Redirect should only happen when bringing the user back to the login screen. Maybe a spinner or some other loading indicator should be shown while logout request is being processed. 

**Screenshots**

https://user-images.githubusercontent.com/32746338/121457256-3814e600-c95c-11eb-8391-d5885ac66641.mov

**Information about your Metabase Installation:**

Current master (0.39.4) on Chrome & Firefox on MacOS Big Sur

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/13082

The previous way of logging in didn't load localization for users with non-default locale (different to the one specified in admin settings). This PR fixes the issue by reloading the browser page, mostly to use the same code to load localization and to avoid weird edge cases. 

The PR also tweaks how logging out is performed to avoid an extra redirect for the user. Please note that `/auth/logout` URL is still available and it can be used for embedding purposes.

How to test:
- Go to admin settings and set the site locale, e.g. `English`.
- Go to account details for the current user and set another locale, e.g. `German`.
- Log out and log in again as the same user.
- Make sure that the app is localized in the language chosen in account details.

### CODE REVIEW

**Review by flamber — COMMENTED** [review]
So while this seem to also fix #16532, it causes a refresh on login even when I haven't defined a locale for the user ("Use site default").

Could it be made so it only causes refresh if the language is not null ("Use site default") and site/user language is different (e.g. Site=English, User=English) ?
I would love if we didn't have to refresh, but I understand the logic we're fighting with.

Is there a way to minimize the refresh? Currently it seems like it's loading the homepage for a split second even when I'm trying to go to a question directly (i.e. `/auth/login?redirect=%2Fquestion%2F123`)

**Review by flamber — APPROVED** [review]
This seems to work great - a lot better than currently, which requires a manual refresh.
I understand most of the code, but please have a FE approve too.

File: frontend/src/metabase/auth/actions.ts [hunk 1]
```diff
@@ -13,6 +16,16 @@ import {
 } from "./analytics";
 import { LoginData } from "./types";
 
+export const REFRESH_LOCALE = "metabase/user/REFRESH_LOCALE";
+export const refreshLocale = createThunkAction(
+  REFRESH_LOCALE,
+  () => async (dispatch: any, getState: () => State) => {
+    const userLocale = getState().currentUser?.locale;
```
Comment (kulyk) [inline frontend/src/metabase/auth/actions.ts:23]: nip: we can use [`getUser` selector here](https://github.com/metabase/metabase/blob/d6c78de7620b728bfb6b44f2ac74ea713931c5d7/frontend/src/metabase/selectors/user.js#L4)

File: frontend/src/metabase/auth/actions.ts [hunk 2]
```diff
@@ -13,6 +16,16 @@ import {
...
+    const siteLocale = Settings.get("site-locale");
```
Comment (kulyk) [inline frontend/src/metabase/auth/actions.ts:24]: Personal observation: I'm so used to seeing `MetabaseSettings`, so `Settings` made me confused for a moment. Should we use `MetabaseSettings` here too for consistency?

**Review by kulyk — APPROVED** [review]
Oh, that works so well! Great work @ranquild  👏

File: frontend/src/metabase/auth/actions.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/auth/actions.ts]: Fixed

File: frontend/src/metabase/auth/actions.ts (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/auth/actions.ts]: Sure, fixed
